crates/filter-parser/src/value.rs

```diff
@@ -80,6 +80,39 @@ pub fn word_exact<'a, 'b: 'a>(tag: &'b str) -> impl Fn(Span<'a>) -> IResult<'a,
     }
 }
 
+/// vector_value          = ( non_dot_word | singleQuoted | doubleQuoted)
+pub fn parse_vector_value(input: Span) -> IResult<Token> {
+    pub fn non_dot_word(input: Span) -> IResult<Token> {
+        let (input, word) = take_while1(|c| is_value_component(c) && c != '.')(input)?;
+        Ok((input, word.into()))
+    }
+
+    let (input, value) = alt((
+        delimited(char('\''), cut(|input| quoted_by('\'', input)), cut(char('\''))),
+        delimited(char('"'), cut(|input| quoted_by('"', input)), cut(char('"'))),
```

Comment:
I guess that's where we can shove our nice error message. Not sure what the span will be though 😬 

```
        delimited(char('\''), cut(|input| quoted_by('\'', input)), cut_with_err(char('\''), expected closing delimiter ')),
        delimited(char('"'), cut(|input| quoted_by('"', input)), cut_with_err(char('"'), expected closing delimiter ")),
```